linked_list/implementation.py

```diff
@@ -1,40 +1,149 @@
+# -*- coding: utf-8 -*-
+
 from .interface import AbstractLinkedList
+from .node import Node
 
+import types
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
-
+    
     def __init__(self, elements=None):
-        pass
-
+        self.iterator = None        # for iterating over nodes
+        self.length = 0
+        
+        # here loop and create a new node for elem in elemts
+        if elements == None or elements == []:
+            self.start = self.end = None
+        else:
+            self.start = new_node = self.end = Node(elements[0])
+            self.length = 1     # for creating self.start
+            
+            for i in elements[1:]:
+                self.length += 1
+                new_node = Node(elements[i-1])
+                self.end.next = new_node        # new node is the new tail
+                self.end = self.end.next        # moving tail to the end
+                self.end.next = None
+                
+    def __iter__(self):
+        # should return something iterable
+        it = self.start       # to set the iterator to start, beginning
+        
+        while it:
+            yield (it.elem)
+            it = it.next
+        
+        raise StopIteration
+    
+        
     def __str__(self):
-        pass
+        result_list_string = None
+        if self.start == None:
+            result_list_string = "[]"
+        else:
+            result_list_string = "["
+            self.iterator = self.start
+            
+            # instead of this use iterators
+            while self.iterator is not None:
+                result_list_string += str(self.iterator.elem) + ", "
+                self.iterator = self.iterator.next
+            
+            result_list_string = result_list_string[:-2] + "]"
+        return result_list_string
 
     def __len__(self):
-        pass
-
-    def __iter__(self):
-        pass
+        return self.length
 
     def __getitem__(self, index):
-        pass
-
+        if self.start == None or index >= self.length:
+            raise IndexError
+        return list(iter(self))[index]
+        
     def __add__(self, other):
-        pass
-
+        # for adding two linked lists, new_list = my_list + LinkedList([1])
+        if self.start == None:
+            self.end = other.start
+        else:
+            self.end.next = other.start
+        
     def __iadd__(self, other):
-        pass
+        # operator.__iadd__(a, b)
+        # a = iadd(a, b) is equivalent to a += b
+        self.__add__(other)
 
     def __eq__(self, other):
-        pass
+        return str(self) == str(other)
```

Comment:
You've made me to be aware of keeping exceptions in mind. Thanks. I'll think that from now on.